### PR TITLE
React draggable

### PR DESCRIPTION
**Pull request by etrobert**

### CODE REVIEW

File: src/App/GraphCanvas/GraphCanvas.tsx [hunk 1]
```diff
@@ -39,75 +38,54 @@ const GraphCanvas = ({ updateGraph }: Props): JSX.Element => {
...
-    <div
-      onWheel={onWheel}
-      id="graph"
-      ref={graphRef}
-      data-pan-x={pan.x}
-      data-pan-y={pan.y}
-      data-zoom={zoom}
+    <DraggableCore
+      onDrag={(e, data) => {
+        if (draggedTasksCount === 0)
```
Comment (justinline) [inline src/App/GraphCanvas/GraphCanvas.tsx:48]: praise: nice implementation :)

File: src/App/GraphCanvas/GraphCanvas.tsx [hunk 2]
```diff
@@ -39,75 +38,54 @@ const GraphCanvas = ({ updateGraph }: Props): JSX.Element => {
...
-            </marker>
-          </defs>
-        </svg>
-        {tasks.map((id) => (
-          <Task key={id} id={id} />
-        ))}
+      {/* The outer div stays in place, receives the dragging events */}
+      <div onWheel={onWheel} id="graph">
+        <p className="Graph__zoom-indicator">
+          {zoom !== 1 && Math.floor(zoom * 100) + "% zoom"}
```
Comment (justinline) [inline src/App/GraphCanvas/GraphCanvas.tsx:55]: question: won't this create a NaN error when `zoom === 1`?

Comment (justinline) [reply src/App/GraphCanvas/GraphCanvas.tsx:55]: (I'm not sure the priority of  `&&` + `+`)

File: src/App/GraphCanvas/GraphCanvas.tsx [hunk 3]
```diff
@@ -39,75 +38,54 @@ const GraphCanvas = ({ updateGraph }: Props): JSX.Element => {
...
+                  className="link-arrow-triangle-path"
+                />
+              </marker>
+            </defs>
+          </svg>
+          {tasks.map((id) => (
+            <Task
+              key={id}
+              id={id}
+              onDragStart={() => setDraggedTasksCount((count) => count + 1)}
```
Comment (justinline) [inline src/App/GraphCanvas/GraphCanvas.tsx:81]: question: Is it possible to drag multiple tasks? If not, this might better be a boolean?

**Comment by justinline** [comment]
Finished 🏁 Just a couple of questions, but nothing blocking. Looks good!

File: src/App/App.tsx [hunk 4]
```diff
@@ -92,7 +92,7 @@ const App = (): JSX.Element => {
         onDelete={onDelete}
       />
 
-      <GraphCanvas updateGraph={updateGraph} />
```
Comment (justinline) [inline src/App/App.tsx:95]: suggestion: We can delete the old methods?

File: src/App/GraphCanvas/GraphCanvas.tsx (hunk 2 above)
Comment (etrobert) [reply src/App/GraphCanvas/GraphCanvas.tsx:55]: I'm pretty sure that `A && B + C = A && (B + C)`

And it seems to work!

I can verify that later at my desk 😊

File: src/App/GraphCanvas/GraphCanvas.tsx (hunk 3 above)
Comment (etrobert) [reply src/App/GraphCanvas/GraphCanvas.tsx:81]: Well I'm not quite sure

I think currently on a touch screen you can drag multiple tasks at the same time

And even if this, the code contains no guarantee that it's not possible, so I chose to play it safe

File: src/App/GraphCanvas/GraphCanvas.tsx (hunk 2 above)
Comment (etrobert) [reply src/App/GraphCanvas/GraphCanvas.tsx:55]: Yes it's the same, prettier even removes the parenthesis if I try to add them

**Pull request merged**